tests/unit/amo/pages/TestAddon.js

```diff
@@ -2965,6 +2968,18 @@ describe(__filename, () => {
       },
     );
 
+    it('does not render the strategic or spotlight badges and correctly renders only the most important badge (RECOMMENDED)', () => {
+      const categories = [LINE, RECOMMENDED, STRATEGIC, SPOTLIGHT];
+      addon.promoted = categories.map((category) => ({
+        category,
+        apps: [clientApp],
+      }));
+      renderWithAddon();
+      expect(screen.getByClassName('PromotedBadge')).toHaveClass(
+        `PromotedBadge--recommended`,
+      );
```

Comment:
This doesn't tell me that other badges aren't shown, maybe make sure that the badge for line extensions isn't in the document.